calour/experiment.py

```diff
@@ -126,8 +126,22 @@ def sparse(self, sparse):
 
     def __repr__(self):
         '''Return a string representation of this object.'''
-        return 'Experiment %s with %d samples, %d features' % (
-            self.description, self.data.shape[0], self.data.shape[1])
+        if self.description:
+            l1 = 'Experiment %s' % self.description
+        else:
+            l1 = 'Experiment'
+        l2 = '-' * len(l1)
+        lines = [l1, l2,
+                 'data dimension: %d samples, %d features' % self.data.shape]
+        try:
+            lines.append('sample IDs: %r' % self.sample_metadata.index)
+        except AttributeError:
+            lines.append('No sample metadata')
+        try:
+            lines.append('feature IDs: %r' % self.feature_metadata.index)
```

Comment:
Same thing. Can be very long for real experiment. maybe show first 5 and last 5 if more than 10 features?